### DISCUSSION TITLE
Is there anything similar for react or vue ?

### DISCUSSION THREAD

**Discussion by padjo** [jonathan-s/django-sockpuppet#98] [Q&A] [answered]
Hi, 
I came across this repository starting from this article https://alistapart.com/article/the-future-of-web-software-is-html-over-websockets/ ,  i know some react and vue and given my skills , i probably should try to deepen them on js/vue before i ever try to jump to django or rails. My question, is there anything like sockpuppet for vue or react or does it even make sense (am sure it does, it's about servr side rendering and thin clients with live updates as far as i understand, also ease of development). Sorry if my question is rambling or possibly off the mark. 

Thanks,
Pat

**Comment by jonathan-s** [comment, marked as answer]
Hi @padjo, 
The ideas behind stimulusreflex / sockpuppet is that you've got a thin client and that you do most of the things server side. There is something called react server components -> https://thoughtbot.com/blog/should-you-react-on-the-server which contains some of the ideas here. That means that your entire stack would be javascript. So perhaps that's what you're looking for?

**Reply by padjo** [reply]
thanks Johathan, 
i'll keep a close eye out on developments for react server. I had thought that graphql/apollo might have been a solution but while it's prob a step in the right direction it's not enough in comparison what rails/stimulus-reflex  and django-sock puppet are doing which seems to be a very thin client using sockets/subscriptions.  https://hackernoon.com/goodbye-redux-26e6a27b3a0b . Thanks for the reply and good luck with this and your other projects. 
Pat
